{{ cookiecutter.repo_name }}/app/src/main/java/{{ cookiecutter.package_dir }}/data/remote/apirequest/ApiResult.kt

```diff
@@ -0,0 +1,39 @@
+package {{ cookiecutter.package_name }}.data.remote.apirequest
+
+import {{ cookiecutter.package_name }}.data.remote.ApiService
+
+/**
+ * Interface to handle different results from a Retrofit API call.
+ * Use this class as the return type for every function placed in [ApiService]
+ *
+ * Usage:
+ * ```
+ *     /* Function getUserData returns an instance of ApiResult */
+ *     apiService.getUserData(id = 101)
```

Comment:
Instead of `apiService` we should have example with `apiRepository` as result is going to be handled at `ViewModel` and their we will be using repository.